File: src/coders/encode.js
const keys = [
  [
    ['a','b','c'],
    ['d','e','f'],
    ['g','h','i']
  ],
  [
    ['j','k','l'],
    ['m','n','o'],
    ['p','q','r']
  ],
  [
    ['s','t','u'],
    ['v','w','x'],
    ['y','z','.']
  ]
]

convertInput=(str)=> {
  let res = ''
  str = str.toLowerCase()
  for(var i=0;i<str.length;i++) {
    if(str[i] === " ") {
      // console.log(res)
      res += ' '
    }
    for(var x=0;x<keys.length;x++) {
      for(var y=0;y<keys.length;y++) {
        for(var z=0;z<keys.length;z++) {
          if(str[i] === keys[x][y][z]) {
            res += ([x+1]+[y+1]+[z+1]+" ")
          }
        }
      }
    }
  }
  return encode(res)
}

encode=(nums)=> {
  let code = ''
  for(var i=0;i<nums.length;i++) {
    if(nums[i] == '1') {
      code += ". "
    } else if (nums[i] == '2') {
      code += ".. "
    } else if (nums[i] == '3') {
      code += '... '
    } else if (nums[i] == ' ' && nums[i+1] == ' ') {
      code += '_ '
    } else if (nums[i] == ' '){
      code += '-'
    }
  }
  return code
}

convertInput("bonjour. Je mapelle victor. Je viens du colorado")

File: src/coders/encode.rb
def convert_input(str)
  keys = [
  [
    ['a','b','c'],
    ['d','e','f'],
    ['g','h','i']
  ],
  [
    ['j','k','l'],
    ['m','n','o'],
    ['p','q','r']
  ],
  [
    ['s','t','u'],
    ['v','w','x'],
    ['y','z','.']
  ]
]
  res = ''
  str = str.downcase
  key_l = str.length
  for j in 0..key_l do
     for x in 0..2 do
        for y in 0..2 do
          for z in 0..2 do
            
            if str[j] == keys[x][y][z]
              xx = x + 1
              yy = y + 1
              zz = z + 1
              res.concat(xx.to_s).concat(yy.to_s).concat(zz.to_s) << " "
            end
          end
        end
     end
     
  end
  encode(res.chop)
end

def encode(nums)
  code = ""
  for i in 0..nums.length do
    if nums[i] == "1"
      code.concat(". ")
    elsif  nums[i] == "2"
      code.concat(".. ")
    elsif nums[i] == "3"
      code.concat("... ")
    else nums[i] == " "
      code.concat("-")
    end
  end
  print code
end

convert_input('hEllo')